specification/frontdoor/resource-manager/Microsoft.Network/preview/2018-08-01-preview/webapplicationfirewall.json

```diff
@@ -0,0 +1,619 @@
+            "schema": {
+              "$ref": "#/definitions/WebApplicationFirewallPolicy"
+            }
+          },
+          "201": {
+            "description": "Created. The request has been fulfilled and a new protection policy has been created.",
+            "schema": {
+              "$ref": "#/definitions/WebApplicationFirewallPolicy"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Creates specific policy": {
+            "$ref": "./examples/WafPolicyCreateOrUpdate.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "DeleteWebApplicationFirewallPolicy"
+        ],
+        "description": "Deletes Policy",
+        "operationId": "Policies_Delete",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "policyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "maxLength": 128,
+            "description": "The name of the resource group."
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted. The request has been accepted for processing and the operation will complete asynchronously."
+          },
+          "204": {
+            "description": "No Content. The request has been accepted but the policy was not found."
+          },
+          "200": {
+            "description": "Delete successful."
+          }
+        },
+        "x-ms-examples": {
+          "Delete protection policy": {
+            "$ref": "./examples/WafPolicyDelete.json"
+          }
+        },        
+        "x-ms-long-running-operation": true
+      }
+    }
+  },
+  "definitions": {
+    "WebApplicationFirewallPolicy": {
+      "description": "Defines web application firewall policy.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/WebApplicationFirewallPolicyPropertiesFormat",
+          "description": "Properties of the web application firewall policy."
+        },
+        "etag": {
+          "type": "string",
+          "description": "Gets a unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ]
+    },
+    "WebApplicationFirewallPolicyPropertiesFormat": {
```

Comment:
Three of your examples defines `resourceState` as should be part of this model, but it's not.

Being that the right thing to do is to model PRECISELY the actual rest API:
- Add resourceState here is this really exists
- Remove it from the examples if this does not really exists (has been removed from API, etc.)

Impacted examples:
- specification/frontdoor/resource-manager/Microsoft.Network/preview/2018-08-01-preview/examples/WafPolicyCreateOrUpdate.json
-  specification/frontdoor/resource-manager/Microsoft.Network/preview/2018-08-01-preview/examples/WafPolicyGet.json
- specification/frontdoor/resource-manager/Microsoft.Network/preview/2018-08-01-preview/examples/WafListPolicies.json